Libraries/Image/RCTImageLoader.m

```diff
@@ -371,7 +371,18 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithTag:(NSString *)imageTag
     // Check for cached response before reloading
     // TODO: move URL cache out of RCTImageLoader into its own module
     NSCachedURLResponse *cachedResponse = [_URLCache cachedResponseForRequest:request];
-    if (cachedResponse) {
+
+    while (cachedResponse) {
+      if ([cachedResponse.response isKindOfClass: [NSHTTPURLResponse class]]) {
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)cachedResponse.response;
+        if (httpResponse.statusCode == 301 || httpResponse.statusCode == 302) {
+          NSURL *redirectURL = [NSURL URLWithString:httpResponse.allHeaderFields[@"Location"]];
+          request = [NSURLRequest requestWithURL: redirectURL];
```

Comment:
Same here. Also can you make sure redirectURL is non-nil before continuing here?